zebra-state/src/response.rs

```diff
@@ -69,11 +69,27 @@ pub enum Response {
     /// specified block hash.
     BlockHash(Option<block::Hash>),
 
+    /// Response to [`Request::KnownBlock`].
+    KnownBlock(Option<KnownBlock>),
+
     #[cfg(feature = "getblocktemplate-rpcs")]
     /// Response to [`Request::CheckBlockProposalValidity`](Request::CheckBlockProposalValidity)
     ValidBlockProposal,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+/// An enum of block stores in the state where a block hash could be found.
+pub enum KnownBlock {
+    /// Block is in the best chain.
+    BestChain,
+
+    /// Block is in a side chain.
+    SideChain,
+
+    /// Block is queued to be validated and committed, or rejected and dropped.
+    Queue,
```

Comment:
I think that we might accidentally label some non-finalized blocks as `Queued`?
This isn't important for this PR, but we should document it to avoid surprises later.

```suggestion
    /// Block is queued to be validated and committed, or rejected and dropped.
    /// Some non-finalized blocks might also appear as `Queued` until the sent blocks have been pruned.
    Queue,
```

Reply:
Oh hang on let me check the rest of the code.

Reply:
I think we can fix this in the code.